src/engine/strat_engine/backstore/setup.rs

```diff
@@ -34,12 +34,14 @@ pub fn find_all() -> StratisResult<HashMap<PoolUuid, HashMap<Device, PathBuf>>>
         match devnode_to_devno(&devnode)? {
             None => continue,
             Some(devno) => {
-                is_stratis_device(&devnode)?.and_then(|pool_uuid| {
+                if let Some((pool_uuid, _)) = StaticHeader::device_identifiers(
+                    &mut OpenOptions::new().read(true).open(&devnode)?,
+                )? {
```

Comment:
It looks like there's a bug in `get_stratis_block_devices` so that this is not safe to do.

`.filter(|x| is_stratis_device(&x).ok().is_some())` This converts the Result to an Option (resulting in an `Option<Option<uuid>>` and then checks if the outermost option is some, but it's only a Stratis device if the interior Option is Some. Try `.filter(|x| is_stratis_device(&x).unwrap_or(None).is_some())`.